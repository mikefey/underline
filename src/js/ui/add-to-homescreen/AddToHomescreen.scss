@import '~colors.scss';

.wrapper {
  display: none;
  height: 100%;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 600;
}

.wrapperShowing {
  composes: wrapper;
  display: block;
}

.background {
  background-color: $white;
  height: 100%;
  opacity: 0.9;
  position: absolute;
  width: 100%;
}

.content {
  background-color: $white;
  box-shadow: 0 0 10px 1px $medium-opacity-black;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-size: 18px;
  font-weight: bold;
  height: 300px;
  left: 50%;
  margin: -150px 0 0 -130px;
  position: absolute;
  top: 50%;
  width: 260px;
}

.text {
  float: left;
  margin-top: 105px;
  padding: 0 20px;
  position: relative;
  width: 100%;
}

.iOSIcon {
  display: inline-block;
  height: 20px;
  margin: 0 -5px;
  position: relative;

  svg {
    height: 100%;
    width: auto;

    path {
      fill: $rose;
    }
  }
}

.androidIcon {
  display: inline-block;
  height: 15px;
  margin: 0 1px 0 5px;
  position: relative;

  svg {
    height: 100%;
    width: auto;

    rect {
      fill: $rose;
    }
  }
}
