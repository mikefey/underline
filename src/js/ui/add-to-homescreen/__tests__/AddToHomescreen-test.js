import test from 'ava';
import React from 'react';
import { renderToString } from 'react-dom/server';

import AddToHomescreen from './../AddToHomescreen.jsx';

const state = {
  show: true,
  os: 'iOS',
};

test('AddToHomescreen component', (t) => {
  const hideAddToHomescreen = () => {};
  const tree = renderToString(
    <AddToHomescreen
      addToHomescreenState={state}
      hideAddToHomescreen={hideAddToHomescreen}
    />,
  );
  t.snapshot(tree);
});
