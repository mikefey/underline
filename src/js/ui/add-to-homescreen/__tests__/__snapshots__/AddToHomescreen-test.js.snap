exports[`AddToHomescreen component 1`] = `"\"<div class=\\\"AddToHomescreen__wrapperShowing___3Bqo97pBz4H7yCg AddToHomescreen__wrapper___2stjrGF9E4DqbAr\\\" data-reactroot=\\\"\\\" data-reactid=\\\"1\\\" data-react-checksum=\\\"-1649735239\\\"><div class=\\\"AddToHomescreen__background___1v2CJNeNPZTVN-k\\\" data-reactid=\\\"2\\\"></div><div class=\\\"AddToHomescreen__content___2fh3KdQla3XXnYH\\\" data-reactid=\\\"3\\\"><button class=\\\"CloseButton__wrapper___2i9TByxUopDp0cZ\\\" data-reactid=\\\"4\\\"><div class=\\\"CloseButton__icon___3Schb-DMASl5XVC\\\" data-reactid=\\\"5\\\"><div class=\\\"CloseButton__iconLineOne___1_bEHFLo8IfP5Fg\\\" data-reactid=\\\"6\\\"></div><div class=\\\"CloseButton__iconLineTwo___3dpb1YfW7QXAzBv CloseButton__iconLineOne___1_bEHFLo8IfP5Fg\\\" data-reactid=\\\"7\\\"></div></div></button><div class=\\\"AddToHomescreen__text___-oZ2IyA4Uo5bG7K\\\" data-reactid=\\\"8\\\"><div data-reactid=\\\"9\\\"><!-- react-text: 10 -->To install, click the<!-- /react-text --><span class=\\\"AddToHomescreen__iOSIcon___fIRNU_lYOstvooa\\\" data-reactid=\\\"11\\\"><svg xmlns=\\\"http://www.w3.org/2000/svg\\\" width=\\\"50\\\" height=\\\"50\\\" viewBox=\\\"0 0 50 50\\\" data-reactid=\\\"12\\\"><path d=\\\"M 25 0.59375 L 24.28125 1.28125 L 16.28125 9.28125 A 1.016466 1.016466 0 1 0 17.71875 10.71875 L 24 4.4375 L 24 32 A 1.0001 1.0001 0 1 0 26 32 L 26 4.4375 L 32.28125 10.71875 A 1.016466 1.016466 0 1 0 33.71875 9.28125 L 25.71875 1.28125 L 25 0.59375 z M 7 16 L 7 17 L 7 49 L 7 50 L 8 50 L 42 50 L 43 50 L 43 49 L 43 17 L 43 16 L 42 16 L 33 16 A 1.0001 1.0001 0 1 0 33 18 L 41 18 L 41 48 L 9 48 L 9 18 L 17 18 A 1.0001 1.0001 0 1 0 17 16 L 8 16 L 7 16 z\\\" color=\\\"#000\\\" data-reactid=\\\"13\\\"></path></svg></span><!-- react-text: 14 --> icon<!-- /react-text --><!-- react-text: 15 --> <!-- /react-text --><span data-reactid=\\\"16\\\">above</span><!-- react-text: 17 --> <!-- /react-text --><!-- react-text: 18 -->and choose &#x27;Add to Home Screen&#x27;<!-- /react-text --></div></div></div></div>\""`;
